lifecycle/index.js

```diff
@@ -156,11 +156,42 @@ export let onMount = (store, initialize) => {
   }
 }
 
-export let onError = (store, listener) =>
-  on(store, listener, ERROR, runListeners => {
+
+export let onAction = (store, listener) =>
+  on(store, listener, ACTION, runListeners => {
+    let errorListeners = {}
+    let endListeners = {}
+    let originAction = store.action
     let originError = store.error
-    store.error = (error, actionName) => runListeners({ error, actionName })
+    let originEnd = store.end
+    store.action = (id, actionName, args) => {
```

Comment:
Adding methods to `store` can lead to dangerous things. User extends store object with custom methods/properties.

But it is just a thinking out loud. Anyway, we already started it and should fix it in another PR.